src/bigquery.ts

```diff
@@ -2391,9 +2371,228 @@ promisifyAll(BigQuery, {
+              wrapIntegers
+            ).valueOf()
+          : BigQuery.int(value)
+        : Number(value);
+      break;
+    }
+    case 'NUMERIC': {
+      value = new Big(value);
+      break;
+    }
+    case 'BIGNUMERIC': {
+      value = new Big(value);
+      break;
+    }
+    case 'RECORD': {
+      value = BigQuery.mergeSchemaWithRows_(schemaField, value, options).pop();
+      break;
+    }
+    case 'DATE': {
+      value = BigQuery.date(value);
+      break;
+    }
+    case 'DATETIME': {
+      value = BigQuery.datetime(value);
+      break;
+    }
+    case 'TIME': {
+      value = BigQuery.time(value);
+      break;
+    }
+    case 'TIMESTAMP': {
+      const pd = new PreciseDate();
+      pd.setFullTime(PreciseDate.parseFull(BigInt(value) * BigInt(1000)));
+      value = BigQuery.timestamp(pd);
+      break;
+    }
+    case 'GEOGRAPHY': {
+      value = BigQuery.geography(value);
+      break;
+    }
+    case 'JSON': {
+      const {parseJSON} = options;
+      value = parseJSON ? JSON.parse(value) : value;
+      break;
+    }
+    case 'RANGE': {
+      value = BigQueryRange.fromSchemaValue_(
+        value,
+        schemaField.rangeElementType!.type!
+      );
+      break;
+    }
+    default:
+      break;
+  }
+
+  return value;
+}
+
+/**
+ * Range class for BigQuery.
+ * A range represents contiguous range between two dates, datetimes, or timestamps.
+ * The lower and upper bound for the range are optional.
+ * The lower bound is inclusive and the upper bound is exclusive.
+ * See https://cloud.google.com/bigquery/docs/reference/standard-sql/lexical#range_literals
+ */
+export class BigQueryRange {
+  elementType?: string;
+  start?: BigQueryTimestamp | BigQueryDate | BigQueryDatetime;
+  end?: BigQueryTimestamp | BigQueryDate | BigQueryDatetime;
+  constructor(value: string | BigQueryRangeOptions, elementType?: string) {
+    if (typeof value === 'string') {
+      if (!elementType) {
+        throw new Error(
+          'invalid RANGE. Element type required when using RANGE literal string.'
+        );
+      }
+      let cleanedValue = value;
+      if (cleanedValue.startsWith('[') || cleanedValue.startsWith('(')) {
+        cleanedValue = cleanedValue.substring(1);
+      }
+      if (cleanedValue.endsWith(')') || cleanedValue.endsWith(']')) {
+        cleanedValue = cleanedValue.substring(0, cleanedValue.length - 1);
+      }
+      const parts = cleanedValue.split(',');
+      if (parts.length !== 2) {
+        throw new Error(
+          'invalid RANGE. See RANGE literal format docs for more information.'
+        );
+      }
+
+      const [start, end] = parts.map((s: string) => s.trim());
+      this.start = this.convertElement_(start, elementType);
+      this.end = this.convertElement_(end, elementType);
+      this.elementType = elementType;
+    } else {
+      const {start, end} = value;
+      if (start && end) {
+        if (typeof start !== typeof end) {
```

Comment:
Will this work in the case when one of `start` and `end` is `undefined`?

Reply:
yeah it works and I added test cases for that.